tests-e2e/fixture/fixture.go

```diff
@@ -0,0 +1,196 @@
+package fixture
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	operation "github.com/redhat-appstudio/managed-gitops/backend-shared/apis/managed-gitops/v1alpha1"
+	managedgitopsv1alpha1 "github.com/redhat-appstudio/managed-gitops/backend/apis/managed-gitops/v1alpha1"
+
+	apps "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/rbac/v1"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	GitOpsServiceE2ENamespace = "gitops-service-e2e"
+)
+
+func EnsureCleanSlate() error {
+
+	policy := metav1.DeletePropagationForeground
+
+	kubeClientSet, err := GetKubeClientSet()
+	if err != nil {
+		return err
+	}
+
+	// Delete the e2e namespace, if it exists
+	err = kubeClientSet.CoreV1().Namespaces().Delete(context.Background(), GitOpsServiceE2ENamespace, metav1.DeleteOptions{PropagationPolicy: &policy})
+	if err != nil && !apierr.IsNotFound(err) {
+		return err
+	}
+
+	// Wait for namespace to delete
+	if err := wait.Poll(time.Second*1, time.Minute*2, func() (done bool, err error) {
+
+		_, err = kubeClientSet.CoreV1().Namespaces().Get(context.Background(), GitOpsServiceE2ENamespace, metav1.GetOptions{})
+		if err != nil {
+			if apierr.IsNotFound(err) {
+				return true, nil
+			} else {
+				return false, err
+			}
+		}
+
+		return false, nil
+	}); err != nil {
+		return err
+	}
+
+	// Create the namespace again
+	_, err = kubeClientSet.CoreV1().Namespaces().Create(context.Background(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{
+		Name: GitOpsServiceE2ENamespace,
+		Labels: map[string]string{
+			"argocd.argoproj.io/managed-by": "gitops-service-argocd",
```

Comment:
can you maybe use the  already existing`const DefaultGitOpsEngineSingleInstanceNamespace = "gitops-service-argocd"` ?